### DISCUSSION TITLE
Difference between fonttools and harfbuzz subsettter

### DISCUSSION THREAD

**Discussion by typoman** [fonttools/fonttools#2248] [Q&A] [answered]
Hi all,

It was recently that I realized besides fonttools, harfbuzz also has a font subsetter. This made me confused as to why two subsetters are maintained by the same group of developers. Does any of these subsetters have an advantage over the other?

Thanks,
Bahman

**Comment by anthrotype** [comment]
https://docs.google.com/document/d/1vDbMhZFkj52F1YbG7X7KeHCc1VUB7hRHgsx4rYwBQls/edit?usp=sharing

**Reply by typoman** [reply]
Thank you Cosimo! if I summarize it, it's mainly to gain performance using C++ and fontTools is older. I thought harfbuzz is more tested since it's being actively developed recently.

**Reply by chrissimpkins** [reply]
Any differences in the vf->static instantiation functionality between the harfbuzz subsetter and varLib.mutator?

**Reply by anthrotype** [reply]
Instancing is not supported yet, but it is planned and currently being worked on

**Reply by chrissimpkins** [reply]
Thanks Cosimo!  Good to hear that.

**Comment by behdad** [comment, marked as answer]
> why two subsetters are maintained by the same group of developers.

Fonttools one always will be more feature-rich, since it's much easier to implement things in Python. The HarfBuzz one is much faster...

**Reply by typoman** [reply]
Thank you Behdad!
